lifecycle/index.d.ts

```diff
@@ -0,0 +1,35 @@
+import type { WritableStore, ReadableStore } from '../atom/index.js'
+
+type Store<Data> = WritableStore<Data> | ReadableStore<Data>
+
+export function onSet<Data>(
+  store: Store<Data>,
+  handler: (payload: {
+    original: unknown[]
+    shared: any
+    stop(): void
+    abort(): void
+  }) => void
+)
+
+export function onChange<Data>(
+  store: Store<Data>,
+  handler: (payload: {
+    original: unknown[]
+    shared: any
+    stop(): void
+    abort(): void
+  }) => void
+)
+
+export function onCreate<Data>(
+  store: Store<Data>,
+  handler: (payload: { shared: any }) => void
```

Comment:
We can avoid `any` by asking developer to clarify what previous event handlers put there:

```ts
export function onCreate<Shared = never>(
  store: Store,
  handler: (payload: { shared: Shared }) => void
)
```